src/server/memory_cmd.cc

```diff
@@ -137,6 +136,9 @@ void MemoryCmd::Run(CmdArgList args) {
     shard_set->pool()->Await([](auto* pb) {
       mi_heap_collect(ServerState::tlocal()->data_heap(), true);
       mi_heap_collect(mi_heap_get_backing(), true);
+#ifdef __GLIBC__
+      malloc_trim(0);  // trims the memory (reduces RSS usage) from the malloc allocator.
+#endif
```

Comment:
We should wrap this in a function, as we call `mi_heap_collect()` directly in `FLUSHALL` and we won't be calling `malloc_trim()`